File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/PyCheckiOSettings.java
package com.jetbrains.edu.python.learning.checkio;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import com.intellij.util.xmlb.annotations.Tag;
import com.intellij.util.xmlb.annotations.Transient;
import com.jetbrains.edu.learning.checkio.account.CheckiOAccount;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(name = "PyCheckiOSettings", storages = @Storage("other.xml"))
public class PyCheckiOSettings implements PersistentStateComponent<PyCheckiOSettings> {
  @Nullable
  @Tag("PyAccount")
  private CheckiOAccount myCheckiOAccount;

  @Nullable
  @Override
  public PyCheckiOSettings getState() {
    return this;
  }

  @Override
  public void loadState(@NotNull PyCheckiOSettings settings) {
    XmlSerializerUtil.copyBean(settings, this);
  }

  @NotNull
  public static PyCheckiOSettings getInstance() {
    return ServiceManager.getService(PyCheckiOSettings.class);
  }

  @Nullable
  @Transient
  public CheckiOAccount getAccount() {
    return myCheckiOAccount;
  }

  @Transient
  public void setAccount(@Nullable CheckiOAccount account) {
    myCheckiOAccount = account;
  }
}

File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/PyCheckiOSettings.kt
package com.jetbrains.edu.python.learning.checkio

import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.openapi.components.service
import com.jetbrains.edu.learning.checkio.CheckiOSettingsBase
import com.jetbrains.edu.learning.checkio.connectors.CheckiOOAuthConnector
import com.jetbrains.edu.python.learning.checkio.connectors.PyCheckiOOAuthConnector

private const val SERVICE_NAME = "PyCheckiOSettings"

@State(name = SERVICE_NAME, storages = [Storage("other.xml")])
class PyCheckiOSettings : CheckiOSettingsBase() {

  override val serviceName: String
    get() = SERVICE_NAME
  override val checkiOOAuthConnector: CheckiOOAuthConnector
    get() = PyCheckiOOAuthConnector

  companion object {
    fun getInstance(): PyCheckiOSettings = service()
  }
}
